port.js

```diff
@@ -126,15 +126,16 @@ Port.prototype.init = function init() {
     this.logFactory && (this.log = this.logFactory.createLog(this.config.logLevel, {name: this.config.id, context: this.config.type + ' port'}, this.config.log));
 
     if (this.config.metrics !== false && this.bus && this.bus.config.implementation && this.bus.performance) {
-        this.counter = function initCounters(type, code, name) {
-            return this.bus.performance.register((this.bus.performance.config.id || this.bus.config.implementation) + '_' +
-                (this.config.metrics || this.config.id), type, code, name);
+        var baseCounterName = (this.bus.performance.config.id || this.bus.config.implementation) + '_' + (this.config.metrics || this.config.id);
+        this.counter = function initCounters(fieldType, fieldCode, fieldName) {
+            return this.bus.performance.register(baseCounterName, fieldType, fieldCode, fieldName);
         }.bind(this);
         this.msgSent = this.counter('counter', 'ms', 'Messages sent');
         this.msgReceived = this.counter('counter', 'mr', 'Messages received');
         this.activeExecCount = this.counter('gauge', 'ae', 'Active exec count');
         this.activeSendCount = this.counter('gauge', 'as', 'Active send count');
         this.activeReceiveCount = this.counter('gauge', 'ar', 'Active receive count');
+        this.timeTaken = this.bus.performance.register(baseCounterName + '_timeTaken', 'average', 'tt', 'Time taken', 'tagged');
```

Comment:
We need to check if we have the expected version of the performance port, before creating this counter. Think about exporting some information in the performance port, that gives us confidence we can create the tagged counter.